integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala

```diff
@@ -33,7 +33,7 @@ object IntersectionTypeMirror {
     def rec(tpe: TypeRepr): TypeRepr = {
       tpe.dealias match
         case AndType(left, right) => concatTypes(rec(left), rec(right))
-        case t                    => prependTypes(t, TypeRepr.of[EmptyTuple])
+        case t                    => prependTypes(tpe, TypeRepr.of[EmptyTuple])
```

Comment:
do you know *why* you have to pass the aliased type here? if so, maybe it's worth a comment?

Reply:
Yes, I mentioned it in the PR description, but I can add a  more detailed explanation here as a comment.

Reply:
Ah, I forgot to read the description. Yeah, in the code would be much better, for future generations :)